### PR TITLE
hw 27.4 solution

### PR DESCRIPTION
**Pull request by VBortnyk**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 1]
```diff
@@ -47,4 +35,17 @@ public MovieSession add(MovieSession movieSession) {
...
         }
     }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query query = session.createQuery(
+                    "FROM MovieSession WHERE  showTime > :start  AND showTime < :end");
+            query.setParameter("start", date.atStartOfDay());
+            query.setParameter("end", date.atStartOfDay().plusDays(1).minusSeconds(1));
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java:45]:
```suggestion
            query.setParameter("end", date.atTime(LocalTime.MAX));
```
try this one

File: src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public Order create(List<Ticket> tickets, User user) {
+        Order order = new Order(tickets, user);
+        ShoppingCart shoppingCart = shoppingCartDao.getByUser(user);
+        order = orderDao.create(order);
+        shoppingCart.getTickets().clear();
+        shoppingCartDao.update(shoppingCart);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java:27]: you should have shoppingCartService.clear() for that

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java (hunk 1 above)
Comment (VBortnyk) [reply src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java]: done

File: src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java (hunk 2 above)
Comment (VBortnyk) [reply src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java]: done

**Pull request merged**